src/main/java/com/rabbitmq/http/client/ReactorNettyClient.java

```diff
@@ -114,20 +117,27 @@ public ReactorNettyClient(String url, String username, String password) {
     }
 
     public ReactorNettyClient(String url, String username, String password, ReactorNettyClientOptions options) {
-        if (url.endsWith("/")) {
-            rootUrl = url.substring(0, url.lastIndexOf("/"));
-        } else {
-            rootUrl = url;
-        }
         objectMapper = options.objectMapper() == null ? createDefaultObjectMapper() : options.objectMapper().get();
 
-        URI uri = URI.create(url);
         client = options.client() == null ?
-            HttpClient.create(httpOptions -> httpOptions.host(uri.getHost()).port(uri.getPort())) : options.client().get();
+            HttpClient.create().baseUrl(url) : options.client().get();
 
         this.token = options.token() == null ? createBasicAuthenticationToken(username, password) : options.token();
 
-        this.errorHandler = options.errorHandler() == null ? ReactorNettyClient::handleError : options.errorHandler();
+        if (options.onResponseCallback() == null) {
+            this.onResponseCallback = (response, connection) -> {
+                if (response.method() == HttpMethod.GET) {
+                    if (response.status().code() >= 500) {
+                        throw new HttpServerException(response.status().code(), response.status().reasonPhrase());
```

Comment:
we might introduce an extra API in the HttpClient builder to make it possible again. In general for generic REST use cases we don't want to do that since we had issues with payload being dropped in the past.
I need to double check if throw works as expected here (it should).

Reply:
There are tests checking this works. I understand this seems awkward, but throwing exception for 4xx and 5xx is something pretty common.
Note this default behavior sticks to what it was with Reactor Netty 0.7 and is pluggable.

Reply:
@smaldini After the upgrade to Reactor Netty 0.8.0.RELEASE, exceptions thrown on the response callback are no longer propagated to the caller (they were in M3). Is it expected (fix, regression)?

Reply:
@smaldini Couldn't spot anything obvious in the [`0.8.0.M3`-`0.8.0.RELEASE` diff](https://github.com/reactor/reactor-netty/compare/v0.8.0.M3...v0.8.0.RELEASE) so it might be something between Reactor Core `3.2.0.RC1` and `3.2.0.RELEASE`.

cc @simonbasle

Reply:
I don't expect anything in core to influence that behavior, but I might be wrong. @violetagg can you have a look from reactor-netty's perspective though?